File: xander/mta.rb
require 'pry'

lines = {	
	"LINE N" => ["8th", "Union Square", "23rd", "28th", "34th", "Times Square"],
	"LINE L" => ["1st", "3rd", "Union Square", "6th", "8th"],
	"LINE 6" => ["Astor Place", "Union Square", "23rd", "28th", "33rd", "Grand Central"]
}

puts lines.keys
print "Type which line are you on?: "
current_line = gets.chomp

puts lines[current_line]
print "Which station are you at?: "
current_station = gets.chomp

puts lines.keys
print "What is your destination line?: "
destination_line = gets.chomp

puts lines[destination_line]
print "What is your destination station?: "
destination_station = gets.chomp

# calculating the number of stops between current and destination station

stops_to_union_square = (lines[current_line].index(current_station) - lines[current_line].find_index("Union Square")).abs
stops_from_union_square = (lines[destination_line].index(destination_station) - lines[destination_line].find_index("Union Square")).abs
number_of_stops = (stops_to_union_square + stops_from_union_square)

# This is the output for the user

puts "Thank you for using Crappy Subway App"
puts "Your travel directions are as follows: "
puts "You are travelling from #{current_station} station on #{current_line} to #{destination_station} station on #{destination_line}."

# this stipulates if the user needs to change lines

if current_line == destination_line
	puts "This route does not require changing trains. Please remain on #{current_line}."
else
	puts "This route requires you to change trains at Union Square to #{destination_line}."
end

puts "The total number of stations you will pass before you arrive at #{destination_station} station on #{destination_line} is #{ (number_of_stops - 1) }."

File: tom/mta_js/mta.js
var lines = {
    'N': ['Times Square', '34th', '28th', '23rd', 'Union Square', '3rd', '1st'],
    'L': ['8th', '6th', 'Union Square', '3rd', '1st'],
    '6': ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
};

var currentLine, currentStation, destinationLine, destinationStation;

do
  currentLine = prompt("Which line are you on?");
while (lines[currentLine] === undefined)

do
  currentStation = prompt("Which station are you at?");
while (lines[currentLine].indexOf(currentStation) === -1);

do
  destinationLine = prompt("What is your destination line?");
while (lines[destinationLine] === undefined)

do
  destinationStation = prompt("What is your destination station?");
while (lines[destinationLine].indexOf(destinationStation) === -1);

// covers the 'leaving from US but going to a different line or vice versa' error
if (currentStation === "Union Square") {
  currentLine = destinationLine;
}
if (destinationStation === "Union Square") {
  destinationLine = currentLine;
}

var currentStationIndex = lines[currentLine].indexOf(currentStation)
var destinationStationIndex = lines[destinationLine].indexOf(destinationStation)
var currentLineUSIndex = lines[currentLine].indexOf("Union Square")
var destinationLineUSIndex = lines[destinationLine].indexOf("Union Square")

var totalStops;
if (currentLine === destinationLine) {
  totalStops = Math.abs(currentStationIndex - destinationStationIndex);
} else {
  var stopsToUnionSquare = Math.abs(currentStationIndex - currentLineUSIndex);
  var stopsFromUnionSquare = Math.abs(destinationStationIndex - destinationLineUSIndex);
  totalStops = stopsToUnionSquare + stopsFromUnionSquare;
}

if (totalStops === 0) {
  console.log("You're already there, dummy.");
} else {

  console.log("Total stops: " + totalStops);
  // single-line journey
  if (currentLine == destinationLine) {
    // if going forwards
    console.log("Starting at " + currentStation);
    if (currentStationIndex < destinationStationIndex) {
      for (var i = currentStationIndex + 1; i < destinationStationIndex; i++) {
        console.log("You will pass " + lines[currentLine][i]);
      }
    } else {
      // else if going backwards
      for (var i = currentStationIndex - 1; i > destinationStationIndex; i--) {
        console.log("You will pass " + lines[currentLine][i]);
      }
    }
    // 2 leg journey
  } else {
    console.log("Starting at " + currentStation);
    // 1st leg forwards and backwards
    if (currentStationIndex < currentLineUSIndex) {
      for (var i = currentStationIndex + 1; i < currentLineUSIndex; i++) {
        console.log("You will pass " + lines[currentLine][i]);
      }
    } else {
      for (var i = currentStationIndex - 1; i > currentLineUSIndex; i--) {
        console.log("You will pass " + lines[currentLine][i]);
      }
    }

    console.log("Change at Union Square");

    // 2nd leg forwards and backwards
    if (destinationLineUSIndex < destinationStationIndex) {
      for (var i = destinationLineUSIndex + 1; i < destinationStationIndex; i++) {
        console.log("You will pass " + lines[destinationLine][i]);
      }
    } else {
      for (var i = destinationLineUSIndex - 1; i > destinationStationIndex; i--) {
        console.log("You will pass " + lines[destinationLine][i]);
      }
    }
  }
  console.log("You will end you journey at " + lines[destinationLine][destinationStationIndex]);
}
